jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/constraint/ServiceLoadActivityLevelConstraint.java

```diff
@@ -71,12 +68,22 @@ public ConstraintsStatus fulfilled(JobInsertionContext iFacts, TourActivity prev
             }
 
         }
-        if (newAct instanceof PickupServiceDEPRECATED || newAct instanceof ServiceActivityNEW) {
+
+        if (newAct.getSize().isPositive()) {
             if (!Capacity.addup(newAct.getSize(), futureMaxLoad).isLessOrEqual(iFacts.getNewVehicle().getType().getCapacityDimensions())) {
                 return ConstraintsStatus.NOT_FULFILLED;
             }
         }
-        if (newAct instanceof DeliverServiceDEPRECATED) {
+
+        /*
+         * REMARK - Balage - This negating could be a bottleneck if called too
+         * many times. Has to be mesured. If rational, the activities could
+         * store their size as absolute value as well. (We could rename
+         * getSize() as getCargoChange(), and the absolute value as
+         * getCargoSize(). For positive or zero activities as Service and Pickup
+         * they could refer to the same object.)
+         */
```

Comment:
This is not a problem, only was a notion by me to help the code to work smoother. 
Generally, not all `invert()` call is unimportant. I've simply found that within this constraint it could be avoided. 
The anomaly is, that the code uses size on activities for two purposes: 
* for getting the dimensions of the cargo: this is a positive value
* for getting the cargo change of the vehicle: this is a signed value

The second is even more problematic, because when we need the first one, we need to use instanceof to determine whether the value should be negated or not (even a new `Capacity.abs()` function would be better, because it would encapsulate the problem within Capacity.)

Also I feel it's an overkill to use Builder within addup, substract and invert. It works internally, would create a Builder class each time it is called. 
For PickAndDeliveryExample, these three methods are called  57.770.331 times (!). 

**My proposal: by keeping the immutable behaver, it could access private fields, so it could work directly on them. This could be done quick and transparent.**
(Additional, optional proposal: it is rather a personal preference, but these three operators could be members rather than statics. For me an `newCap = cap1.substract(cap2)` is more natural and OOP than `Capacity.substract(cap1, cap2)`. But this is really subjective,)